mir-ci/mir_ci/display_server.py

```diff
@@ -33,31 +38,50 @@ def wait_for_wayland_display(runtime_dir: str, name: str) -> None:
             return
     raise RuntimeError('Wayland display ' + name + ' did not appear')
 
-class DisplayServer:
-    def __init__(self, command: Command, add_extensions: Tuple[str, ...] = ()) -> None:
-        self.command = command
+class DisplayServer(Benchmarkable):
+    def __init__(self, app: Dependency, add_extensions: Tuple[str, ...] = ()) -> None:
+        self.command: Command = cast(Command, app.command)
+        self.command_type = app.app_type
         self.add_extensions = add_extensions
         # Snaps require the display to be in the form "waland-<number>". The 00 prefix lets us
         # easily identify displays created by this test suit and remove them in bulk if a bunch
         # don't get cleaned up properly.
         self.display_name = 'wayland-00' + str(os.getpid())
 
-    def program(self, command: Command, env: Dict[str, str] = {}) -> Program:
+    def get_cgroup(self) -> Cgroup:
+        if self.server:
+            return self.server.get_cgroup()
+        raise RuntimeError("Attempting to get the cgroup before the server exists")
+    
+    def get_name(self) -> str:
+        if self.server:
+            return self.server.get_name()
+        raise RuntimeError("Attempting to get the name before the server exists")
+
+    def program(
+            self, 
+            command: Command,
+            env: Dict[str, str] = {},
+            app_type: Optional[DependencyType] = None) -> Program:
         return Program(command, env=dict({
                 'DISPLAY': 'no',
                 'QT_QPA_PLATFORM': 'wayland',
                 'WAYLAND_DISPLAY': self.display_name
             },
-            **env)
-        )
+            **env),
+            systemd_slice=app_type == "deb" and f"mirci-{uuid4()}" or None)
```

Comment:
You have `pip`s, too.

```suggestion
            systemd_slice=app_type != "snap" and f"mirci-{uuid4()}" or None)
```